src/inference/src/dev/core_impl.cpp

```diff
@@ -804,12 +804,12 @@ ov::Plugin ov::CoreImpl::get_plugin(const std::string& pluginName) const {
             try {
                 std::vector<ov::Extension::Ptr> ext;
                 desc.extensionCreateFunc(ext);
-                add_extensions_unsafe(ext);
+                add_extensions_unsafe(ext, deviceName);
```

Comment:
As current solution will work but to consider the change to use.
`desc (PluginDescriptor)` to store device extension as vector.
Then vector with extension can be as type to store them in core and device. and on plugin unload the descriptor can be updated to clean the vector.

The get_extension_copy could take core extension vector and loop over plugin descriptors and insert plugin extensions.